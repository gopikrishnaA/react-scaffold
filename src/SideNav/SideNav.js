import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import {
  Links,
  NavItem,
  NavTitle,
  NavLabel,
  NavIcon,
  CustomLink,
  IconSize
} from './SideNavLinks'
import {
  faChevronCircleLeft,
  faChevronCircleRight,
  faTimes,
  faBars
} from 'fa5-pro-solids'
import UserNav from '../TopNav/UserNav/UserNav'

const SideNavContainer = styled.div`
  display: grid;
  grid-template-rows: 56px auto;
  grid-template-areas: ' Logo ' ' Links ';
  z-index: 20;
  bottom: 0px;
  grid-area: SideNav;
  position: absolute;
  height: 100vh;
  color: #ffffff40;
  background: ${p => p.theme.palette && p.theme.palette.background.nav2};
  width: ${p => (p.sideNavisOpen ? '240px' : '60px ')};
  box-shadow: ${p => (p.sideNavisOpen ? '2px 4px 6px rgba(0,0,0,.24)' : 'none')};
  transition: width 185ms;
  transition-property: width;
  transition-delay: 60ms;
  @media (max-width : 768px) {
    display: ${p => (p.sideNavisOpen ? 'grid' : 'none')}
  }
`

const Grid = styled.div`
  display: grid;
  height: 100vh;
  grid-template-columns: auto 1fr;
  grid-template-rows: 56px 1fr;
  grid-template-areas:  
  ' SideNav ContentArea ';
`

const Image = styled.img`
width: 25px;
margin-right: 10px;
`

const BottomNavItem = styled(NavItem)`
position: absolute;
bottom: 0;
`

const RightIcon = styled(IconSize)`
position: absolute;
bottom: 10px;
right: 10px;
`

const MenuIcon = styled(NavItem)`
  z-index: 20;
  cursor: pointer;
  color: #000;
  bottom: 50px;
  @media (min-width : 769px) {
    display: none;
  }
`

const CloseIcon = styled.div`
  @media (max-width : 768px) {
    height: 50px;
    display: grid;
    width: 100%;
    cursor: pointer;
    align-items: end;
    position: absolute;
    color: #000;
    justify-content: end;
    left: 30px;
    top: 0
  }
`

const CloseIconeSize = styled(IconSize)`
  display: none;
  @media (max-width : 768px) {
    display: block;
  }
`

const UserNavItem = styled(NavItem)`
  display: none;
  @media (max-width : 768px) {
    display: block;
    margin-top: 50px;
  }
`
const BottomLine = styled.div`
  margin: 0 10px;
  border-bottom: 1px solid #fff;
`

export default class SideNav extends React.Component {
  state = {
    sideNavisOpen: false
  }

  toggleNav = () => {
    this.setState({ sideNavisOpen: !this.state.sideNavisOpen })
  }

  render () {
    const { sideNavLinks = [],
      appName = '',
      appLogo,
      userAvatar,
      firstName,
      lastName,
      userProfession,
      settingsIcon,
      isFromSideNav } = this.props
    return (
      <Grid>
        <MenuIcon onClick={this.toggleNav}>
          <IconSize icon={faBars} />
        </MenuIcon>
        <SideNavContainer sideNavisOpen={this.state.sideNavisOpen}>
          <NavTitle>
            <Image src={appLogo} />
            {this.state.sideNavisOpen && <span>{appName}</span>}
          </NavTitle>
          <UserNavItem>
            <UserNav
              isFromSideNav={isFromSideNav}
              userAvatar={userAvatar}
              firstName={firstName}
              lastName={lastName}
              userProfession={userProfession}
              settingsIcon={settingsIcon} />
            <BottomLine />
          </UserNavItem>
          <Links>
            {sideNavLinks.map((link, index) => (
              <NavItem key={index} onClick={this.toggleNav}>
                <CustomLink>
                  <NavIcon>
                    <IconSize icon={link.icon} />
                  </NavIcon>
                  {this.state.sideNavisOpen &&
                    <NavLabel to={link.path}>
                      {link.title}
                    </NavLabel>}
                </CustomLink>
              </NavItem>
            ))}
            <BottomNavItem>
              <CustomLink>
                {!this.state.sideNavisOpen
                  ? <NavIcon onClick={this.toggleNav}>
                    <IconSize icon={faChevronCircleRight} />
                  </NavIcon>
                  : <NavIcon onClick={this.toggleNav}>
                    <RightIcon icon={faChevronCircleLeft} />
                  </NavIcon>}
              </CustomLink>
            </BottomNavItem>
          </Links>
          <CloseIcon onClick={this.toggleNav}>
            <CloseIconeSize icon={faTimes} />
          </CloseIcon>
        </SideNavContainer>
      </Grid>
    )
  }
}

SideNav.propTypes = {
  sideNavLinks: PropTypes.array,
  appLogo: PropTypes.string,
  appName: PropTypes.string,
  userAvatar: PropTypes.string,
  firstName: PropTypes.string,
  lastName: PropTypes.string,
  userProfession: PropTypes.string,
  settingsIcon: PropTypes.object,
  isFromSideNav: PropTypes.bool
}
