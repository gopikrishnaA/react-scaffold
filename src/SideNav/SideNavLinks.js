import styled from 'styled-components'
import ButtonBase from '@material-ui/core/ButtonBase'
import FontAwesomeIcon from '@fortawesome/react-fontawesome'
import {Link} from 'react-router-dom'

export const Links = styled.div`
  display: grid;
  grid-area: Links;
  grid-template-rows: 50px 50px 50px 50px 50px 50px;
  align-items: center;
  @media (max-width : 768px) {
    margin-top: 60px;
  }
`

export const NavItem = styled(ButtonBase)`
  height: 50px;
  display: grid;
  width: 100%;
  padding: 5px;
  grid-template-columns: 50px 1fr;
  cursor: pointer;
  align-items: center;
  justify-content: center;
  position: relative;
  color: #fff;
  &:hover {
    color: ${p => p.theme.palette && p.theme.palette.text.alt1};
    background: rgba(0, 0, 0, 0.2);
  }
`

export const NavTitle = styled.div`
letter-spacing: .05em;
  color: ${p => p.theme.palette && p.theme.palette.text.alt1};
  font-weight: 400;
  position: absolute;
  left: 20px;
  top: 22px;

  span {
    font-weight: 600;
    position: fixed;
    padding-top: 5px;
  }
`

export const NavLabel = styled(Link)`
  letter-spacing: .08em;
  font-weight: 300;
  text-transform: uppercase;
  font-size:14px;
  position: absolute;
  left: 60px;
  top: 20px;
  text-decoration: none !important;
  color: ${p => p.theme.palette && p.theme.palette.text.alt1};
`

export const NavIcon = styled.div`
  width: 50px;
  padding: 0;
  display: grid;
  justify-content: center;
  align-items: center;
`
export const IconSize = styled(FontAwesomeIcon)`
  font-size: 2.2em;
`
export const CustomLink = styled.a`
  transition: 0.3s all ease-in-out;
  padding: 10px 0;

  &:link {
    color: ${p => p.theme.palette && p.theme.palette.text.alt2};
  }
  &:visited {
    color: ${p => p.theme.palette && p.theme.palette.text.alt2};
  }
  &:hover {
    color: ${p => p.theme.palette && p.theme.palette.text.alt1};
    transition: 0.3s all ease-in-out;
  }
`
