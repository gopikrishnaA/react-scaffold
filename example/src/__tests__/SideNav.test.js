import React from 'react'
import renderer from 'react-test-renderer'
import SideNav from 'connect-nav/lib/SideNav'
import data from '../static/data'
it('renders correctly', () => {
  const tree = renderer
    .create(
      <SideNav
        appLogo={data.appLogo}
        appName={data.appName}
        sideNavLinks={data.sideNavLinks}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly', () => {
  const tree = renderer.create(<SideNav />).toJSON()
  expect(tree).toMatchSnapshot()
})
