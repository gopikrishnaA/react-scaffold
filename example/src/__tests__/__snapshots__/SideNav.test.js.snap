// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-htoDjs dvzNqK"
>
  <button
    className="MuiButtonBase-root-1 sc-gqjmRU cMAnHd sc-bwzfXH hoTZzz"
    disabled={undefined}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-bars fa-w-14 sc-EHOje iprojn"
      data-icon="bars"
      data-prefix="fas"
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <span
      className="MuiTouchRipple-root-4"
    />
  </button>
  <div
    className="sc-gzVnrw bRSXgo"
  >
    <div
      className="sc-htpNat dIzBcT"
    >
      <img
        className="sc-dnqmqq jilVsi"
        src="logo.svg"
      />
    </div>
    <div
      className="sc-bdVaJa jwJPxy"
    >
      <button
        className="MuiButtonBase-root-1 sc-bwzfXH hoTZzz"
        disabled={undefined}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <a
          className="sc-bZQynM hjskkH"
        >
          <div
            className="sc-ifAKCX jMPaRf"
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-home fa-w-16 sc-EHOje iprojn"
              data-icon="home"
              data-prefix="fal"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M509.8 227.5L448 177.8v-76c0-3.3-2.7-6-6-6h-20c-3.3 0-6 2.7-6 6v50.1L276.1 39.1c-11.7-9.5-28.5-9.5-40.2 0L2.2 227.5c-2.6 2.1-3 5.9-.9 8.4l12.6 15.6c2.1 2.6 5.9 3 8.5.9L64 218.9v229c0 17.7 14.3 32 32 32h116c6.6 0 12-5.4 12-12V335.8l64 .3v132.2c0 6.6 5.4 12 12 12l116-.3c17.7 0 32-14.3 32-32V219l41.6 33.5c2.6 2.1 6.4 1.7 8.5-.9l12.6-15.6c2.1-2.6 1.6-6.4-.9-8.5zM416 448l-96 .3V316c0-6.6-5.4-12-12-12l-104-.3c-6.6 0-12 5.4-12 12V448H96V193.1l156.2-126c2.2-1.8 5.3-1.8 7.5 0l156.2 126V448z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </div>
        </a>
        <span
          className="MuiTouchRipple-root-4"
        />
      </button>
      <button
        className="MuiButtonBase-root-1 sc-bwzfXH hoTZzz"
        disabled={undefined}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <a
          className="sc-bZQynM hjskkH"
        >
          <div
            className="sc-ifAKCX jMPaRf"
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-folder-open fa-w-18 sc-EHOje iprojn"
              data-icon="folder-open"
              data-prefix="fal"
              role="img"
              style={Object {}}
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M527.95 224H480v-48c0-26.51-21.49-48-48-48H272l-64-64H48C21.49 64 0 85.49 0 112v288c0 26.51 21.49 48 48 48h385.057c28.068 0 54.135-14.733 68.599-38.84l67.453-112.464C588.24 264.812 565.285 224 527.95 224zM48 96h146.745l64 64H432c8.837 0 16 7.163 16 16v48H171.177c-28.068 0-54.135 14.733-68.599 38.84L32 380.47V112c0-8.837 7.163-16 16-16zm493.695 184.232l-67.479 112.464A47.997 47.997 0 0 1 433.057 416H44.823l82.017-136.696A48 48 0 0 1 168 256h359.975c12.437 0 20.119 13.568 13.72 24.232z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </div>
        </a>
        <span
          className="MuiTouchRipple-root-4"
        />
      </button>
      <button
        className="MuiButtonBase-root-1 sc-bwzfXH hoTZzz"
        disabled={undefined}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <a
          className="sc-bZQynM hjskkH"
        >
          <div
            className="sc-ifAKCX jMPaRf"
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-dollar-sign fa-w-8 sc-EHOje iprojn"
              data-icon="dollar-sign"
              data-prefix="fal"
              role="img"
              style={Object {}}
              viewBox="0 0 256 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M191.9 259.3L73.7 222.2C49.2 214.5 32 189 32 160.3 32 124.8 57.6 96 89 96h73.8c22.2 0 43.3 8.6 60.1 24.5 3.1 2.9 7.8 3.2 11 .3l11.9-10.8c3.4-3.1 3.6-8.4.4-11.6-22.8-22-52.7-34.5-83.3-34.5H144V8c0-4.4-3.6-8-8-8h-16c-4.4 0-8 3.6-8 8v56H89c-49.1 0-89 43.2-89 96.3 0 42.6 26.4 80.6 64.1 92.4l118.2 37.1c24.6 7.7 41.7 33.2 41.7 61.9 0 35.4-25.6 64.3-57 64.3H93.2c-22.2 0-43.3-8.6-60.1-24.5-3.1-2.9-7.8-3.2-11-.3L10.3 402c-3.3 3-3.6 8.4-.3 11.5 22.8 22 52.7 34.5 83.3 34.5H112v56c0 4.4 3.6 8 8 8h16c4.4 0 8-3.6 8-8v-56h23c49.1 0 89-43.2 89-96.3 0-42.5-26.4-80.5-64.1-92.4z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </div>
        </a>
        <span
          className="MuiTouchRipple-root-4"
        />
      </button>
      <button
        className="MuiButtonBase-root-1 sc-bwzfXH hoTZzz"
        disabled={undefined}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <a
          className="sc-bZQynM hjskkH"
        >
          <div
            className="sc-ifAKCX jMPaRf"
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-gem fa-w-18 sc-EHOje iprojn"
              data-icon="gem"
              data-prefix="fal"
              role="img"
              style={Object {}}
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M463.7 0H112.3c-4.2 0-8.1 2.2-10.3 5.8L1.7 168.6c-2.7 4.4-2.2 10.1 1.2 14l276 325.2c4.8 5.6 13.4 5.6 18.2 0l276-325.2c3.4-3.9 3.8-9.6 1.2-14L474 5.8c-2.2-3.6-6.1-5.8-10.3-5.8zm-13.6 36l74.3 124h-83L384.6 36h65.5zM345 36l56.8 124H174.1L231 36h114zm-219.1 0h65.5l-56.8 124h-83l74.3-124zM61.2 192h73L216 384 61.2 192zm112 0h229.5L288 455.8 173.2 192zM360 384l81.8-192h73L360 384z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </div>
        </a>
        <span
          className="MuiTouchRipple-root-4"
        />
      </button>
      <button
        className="MuiButtonBase-root-1 sc-bwzfXH hoTZzz"
        disabled={undefined}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <a
          className="sc-bZQynM hjskkH"
        >
          <div
            className="sc-ifAKCX jMPaRf"
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-medkit fa-w-16 sc-EHOje iprojn"
              data-icon="medkit"
              data-prefix="fal"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M464 96H352V56c0-13.255-10.745-24-24-24H184c-13.255 0-24 10.745-24 24v40H48c-26.51 0-48 21.49-48 48v288c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V144c0-26.51-21.49-48-48-48zM192 64h128v32H192V64zm192 64v320H128V128h256zM32 432V144c0-8.822 7.178-16 16-16h48v320H48c-8.822 0-16-7.178-16-16zm448 0c0 8.822-7.178 16-16 16h-48V128h48c8.822 0 16 7.178 16 16v288zM352 272v32c0 6.627-5.373 12-12 12h-56v56c0 6.627-5.373 12-12 12h-32c-6.627 0-12-5.373-12-12v-56h-56c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h56v-56c0-6.627 5.373-12 12-12h32c6.627 0 12 5.373 12 12v56h56c6.627 0 12 5.373 12 12z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </div>
        </a>
        <span
          className="MuiTouchRipple-root-4"
        />
      </button>
      <button
        className="MuiButtonBase-root-1 sc-bwzfXH hoTZzz"
        disabled={undefined}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <a
          className="sc-bZQynM hjskkH"
        >
          <div
            className="sc-ifAKCX jMPaRf"
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-edit fa-w-18 sc-EHOje iprojn"
              data-icon="edit"
              data-prefix="fal"
              role="img"
              style={Object {}}
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M417.8 315.5l20-20c3.8-3.8 10.2-1.1 10.2 4.2V464c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V112c0-26.5 21.5-48 48-48h292.3c5.3 0 8 6.5 4.2 10.2l-20 20c-1.1 1.1-2.7 1.8-4.2 1.8H48c-8.8 0-16 7.2-16 16v352c0 8.8 7.2 16 16 16h352c8.8 0 16-7.2 16-16V319.7c0-1.6.6-3.1 1.8-4.2zm145.9-191.2L251.2 436.8l-99.9 11.1c-13.4 1.5-24.7-9.8-23.2-23.2l11.1-99.9L451.7 12.3c16.4-16.4 43-16.4 59.4 0l52.6 52.6c16.4 16.4 16.4 43 0 59.4zm-93.6 48.4L403.4 106 169.8 339.5l-8.3 75.1 75.1-8.3 233.5-233.6zm71-85.2l-52.6-52.6c-3.8-3.8-10.2-4-14.1 0L426 83.3l66.7 66.7 48.4-48.4c3.9-3.8 3.9-10.2 0-14.1z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </div>
        </a>
        <span
          className="MuiTouchRipple-root-4"
        />
      </button>
      <button
        className="MuiButtonBase-root-1 sc-iwsKbI cHbNUR sc-bwzfXH hoTZzz"
        disabled={undefined}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <a
          className="sc-bZQynM hjskkH"
        >
          <div
            className="sc-ifAKCX jMPaRf"
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-chevron-circle-right fa-w-16 sc-EHOje iprojn"
              data-icon="chevron-circle-right"
              data-prefix="fas"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8c137 0 248 111 248 248S393 504 256 504 8 393 8 256 119 8 256 8zm113.9 231L234.4 103.5c-9.4-9.4-24.6-9.4-33.9 0l-17 17c-9.4 9.4-9.4 24.6 0 33.9L285.1 256 183.5 357.6c-9.4 9.4-9.4 24.6 0 33.9l17 17c9.4 9.4 24.6 9.4 33.9 0L369.9 273c9.4-9.4 9.4-24.6 0-34z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </div>
        </a>
        <span
          className="MuiTouchRipple-root-4"
        />
      </button>
    </div>
    <div
      className="sc-VigVT hjmNcm"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-times fa-w-11 sc-jTzLTM bhrsiC sc-EHOje iprojn"
        data-icon="times"
        data-prefix="fas"
        role="img"
        style={Object {}}
        viewBox="0 0 352 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`renders correctly 2`] = `
<div
  className="sc-htoDjs dvzNqK"
>
  <button
    className="MuiButtonBase-root-1 sc-gqjmRU cMAnHd sc-bwzfXH hoTZzz"
    disabled={undefined}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-bars fa-w-14 sc-EHOje iprojn"
      data-icon="bars"
      data-prefix="fas"
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <span
      className="MuiTouchRipple-root-4"
    />
  </button>
  <div
    className="sc-gzVnrw bRSXgo"
  >
    <div
      className="sc-htpNat dIzBcT"
    >
      <img
        className="sc-dnqmqq jilVsi"
        src={undefined}
      />
    </div>
    <div
      className="sc-bdVaJa jwJPxy"
    >
      <button
        className="MuiButtonBase-root-1 sc-iwsKbI cHbNUR sc-bwzfXH hoTZzz"
        disabled={undefined}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <a
          className="sc-bZQynM hjskkH"
        >
          <div
            className="sc-ifAKCX jMPaRf"
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-chevron-circle-right fa-w-16 sc-EHOje iprojn"
              data-icon="chevron-circle-right"
              data-prefix="fas"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8c137 0 248 111 248 248S393 504 256 504 8 393 8 256 119 8 256 8zm113.9 231L234.4 103.5c-9.4-9.4-24.6-9.4-33.9 0l-17 17c-9.4 9.4-9.4 24.6 0 33.9L285.1 256 183.5 357.6c-9.4 9.4-9.4 24.6 0 33.9l17 17c9.4 9.4 24.6 9.4 33.9 0L369.9 273c9.4-9.4 9.4-24.6 0-34z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </div>
        </a>
        <span
          className="MuiTouchRipple-root-4"
        />
      </button>
    </div>
    <div
      className="sc-VigVT hjmNcm"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-times fa-w-11 sc-jTzLTM bhrsiC sc-EHOje iprojn"
        data-icon="times"
        data-prefix="fas"
        role="img"
        style={Object {}}
        viewBox="0 0 352 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
</div>
`;
